stable_baselines3/common/torch_layers.py

```diff
@@ -74,7 +74,7 @@ def __init__(self, observation_space: gym.spaces.Box, features_dim: int = 512):
             nn.ReLU(),
             nn.Conv2d(32, 64, kernel_size=4, stride=2, padding=0),
             nn.ReLU(),
-            nn.Conv2d(64, 32, kernel_size=3, stride=1, padding=0),
+            nn.Conv2d(64, 64, kernel_size=3, stride=1, padding=0),
```

Comment:
nice catch :see_no_evil: Please don't tell me that solve your performance issue.

I know where it comes from ... I shouldn't have copy-pasted from https://github.com/ikostrikov/pytorch-a2c-ppo-acktr-gail/blob/master/a2c_ppo_acktr/model.py#L169

Reply:
thinking about that, we need to double check `VecFrameStack`, even though it is the same as SB2.

Reply:
Sadly (luckily? =) ) it did not fix the issues yet. SB3 is still consistently worse in a few of the Atari games I have tested. I am in the process of step-by-step comparisons, will see how that goes.

Reply:
or having that kind of issue: https://github.com/ikostrikov/pytorch-a2c-ppo-acktr-gail/commit/84a7582477fb0d5c82ad6d850fe476829dddd2e1

btw, is it better now with `OMP_NUM_THREADS=1` w.r.t. fps? (maybe you should write in the comment the current stand of SB2 vs SB3)

Reply:
one thing that may change is the optimizer implementation and default parameters, for the initialization, I think (at least I tried) to reproduce what was done in SB2.

Reply:
I remember creating the correct CNN here: https://github.com/hill-a/stable-baselines/issues/372

```python
class PyTorchCnnPolicy(nn.Module):
    def __init__(self):
        super(PyTorchCnnPolicy, self).__init__()
        self.conv1 = nn.Conv2d(in_channels=4, out_channels=32, kernel_size=8, stride=4, padding=0, bias=True)
        self.conv2 = nn.Conv2d(in_channels=32, out_channels=64, kernel_size=4, stride=2, padding=0, bias=True)
        self.conv3 = nn.Conv2d(in_channels=64, out_channels=64, kernel_size=3, stride=1, padding=0, bias=True)            
        self.fc1 = nn.Linear(3136, 512)
        self.fc2 = nn.Linear(512, 4)      
        self.relu = nn.ReLU()
        self.out_activ = nn.Softmax(dim=1)

    def forward(self, x):
        x = self.relu(self.conv1(x))
        x = self.relu(self.conv2(x))
        x = self.relu(self.conv3(x))
        # shape before flattening
        # tf: (?, 7, 7, 64)
        # pytorch: [1, 64, 7, 7]
        x = x.permute(0, 2, 3, 1).contiguous()
        x = x.view(x.size(0), -1)
        x = self.relu(self.fc1(x))
        x = self.fc2(x)
        x = self.out_activ(x)                  
        return x
```

I could load model from the zoo of SB2 into pytorch.

Reply:
>Edit: Ah yes, stacking on the wrong channels?

Oh, I did not think about that... Should be fine if it is called before the `VecTranspose` though.

Reply:
> btw, is it better now with OMP_NUM_THREADS=1 w.r.t. fps? (maybe you should write in the comment the current stand of SB2 vs SB3)

I still need to test this bit more throughout, but with small/quick envs it makes all the difference (200fps -> 1000fps), but with larger ones I am still uncertain.

One major change in parameters is the use of `batch_size=64` rather than `nminibatches=4` in PPO. Using such small batch-size made things very slow FPS-wise, but in some cases sped up the learning. I will focus more on these running-speed things in an another PR.

Reply:
my question was more what is the fps we want to reach? (what did you have with SB2?)

>One major change in parameters is the use of batch_size=64 rather than nminibatches=4

Yes, the issue was that `nminibatches` lead to different mini-batchsize depending on the number of environments

Reply:
> my question was more what is the fps we want to reach? (what did you have with SB2?)

Hmm I do not have conclusive numbers just yet because I have been running many experiments on same system and can not guarantee fair comparisons, but I _think_ PyTorch variants are about 10% slower with Atari games and 25% slower on toy environments. The latter required the OMP_NUM_THREADS tuning. This sounds reasonable, given the non-compiled nature of PyTorch and the fact the code has not been optimized much yet.

> Yes, the issue was that nminibatches lead to different mini-batchsize depending on the number of environments

Ah alright. I will write big notes about this on the "moving from stable-baselines" docs :)

Reply:
> One major change in parameters is the use of `batch_size=64` rather than `nminibatches=4` in PPO. Using such small batch-size made things very slow FPS-wise, but in some cases sped up the learning. I will focus more on these running-speed things in an another PR.

I would like to add that we may be able to gain a non minuscule speedup by delaying buffer stores and adding the data in batches.

Reply:
I started documenting the migration here ;)
https://github.com/DLR-RM/stable-baselines3/pull/123

>I would like to add that we may be able to gain a non minuscule speedup by avoiding single data stores but instead storing a while batch at once.

?

Reply:
Mistyped, I meant that if we store a whole batch at once, we should get a sizeable speedup over storing one transition at a time.

Reply:
still not sure what you mean...